src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -273,28 +313,75 @@ def get(self, request: Request, project: Project) -> Response:
-                            service=current_config["config"]["provider"]["key"],
-                            branch=current_config["config"]["defaultBranch"],
-                            path=current_config["outcome"]["sourcePath"],
+                        ref = (
+                            ctx["commit_id"]
+                            if ctx["commit_id"]
+                            else current_config["config"]["defaultBranch"]
                         )
-                        if lineCoverage and codecovUrl:
-                            result["codecov"] = {
-                                "lineCoverage": lineCoverage,
-                                "coverageUrl": codecovUrl,
-                                "status": 200,
-                            }
-
-                            if not codecov_cache:
-                                cache.set(cache_key.format(project_id=project.id), True, 3600)
+
+                        # Get commit sha from Git blame
+                        codecov_commit_sha_from_git_blame_enabled = features.has(
+                            "organizations:codecov-commit-sha-from-git-blame",
+                            project.organization,
+                            actor=request.user,
+                        )
+                        gh_integrations = integrations.filter(provider="github")
+                        should_get_commit_sha = (
+                            codecov_commit_sha_from_git_blame_enabled
+                            and len(gh_integrations) > 0
+                            and ref == current_config["config"]["defaultBranch"]
+                        )
+
+                        if should_get_commit_sha:
+                            try:
+                                integration_installation = gh_integrations[0].get_installation(
+                                    organization_id=project.organization_id
+                                )
+                                line_no = ctx.get("line_no")
+                                if line_no:
+                                    ref = self.get_latest_commit_sha_from_blame(
+                                        integration_installation,
+                                        int(line_no),
+                                        filepath,
+                                        current_config["repository"],
+                                        current_config["config"]["defaultBranch"],
+                                    )
+                            except Exception:
+                                logger.exception(
+                                    "Failed to get commit sha from git blame, pending investigation. Continuing execution."
+                                )
+
+                        if not codecov_commit_sha_from_git_blame_enabled or (
+                            codecov_commit_sha_from_git_blame_enabled
+                            and ref != current_config["config"]["defaultBranch"]
```

Comment:
Because I'm a human that can't do logic!

`codecov_commit_sha_from_git_blame_enabled = True`
not True or (True and Foo) -> False or True and Foo -> True and Foo -> Foo

`codecov_commit_sha_from_git_blame_enabled = False`
not False or (False and Foo) -> True or False -> True

I wonder if this could simply be `if ref != current_config["config"]["defaultBranch"]:`
We only check the coverage if we have a commit.

@snigdhas could you please verify this logic? I'm failing here.

Reply:
I think you're correct - it could be simplified to `if not codecov_commit_sha_from_git_blame_enabled or ref != current_config["config"]["defaultBranch"]`

Good catch, thank you!

Reply:
Logic checks out, but let's see if we can refactor this in a way that's easier to parse overall with the `ref != current_config["config"]["defaultBranch"]`. I can see future-me being confused about this once I've forgotten the context